### PR TITLE
[k8s-extension] Release Public Preview Version v0.2.0

### PR DESCRIPTION
**Pull request by jonathan-innis**
Release the public preview version of k8s-extension v0.2.0
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls help to support the release?

File: src/k8s-extension/azext_k8s_extension/custom.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    if configuration_settings is not None and configuration_settings_file is not None:
+        raise MutuallyExclusiveArgumentError(
+            'Error! Both configuration-settings and configuration-settings-file cannot be provided.'
+        )
+
+    if configuration_protected_settings is not None and configuration_protected_settings_file is not None:
+        raise MutuallyExclusiveArgumentError(
+            'Error! Both configuration-protected-settings and configuration-protected-settings-file '
+            'cannot be provided.'
+        )
```
Comment (fengzhou-msft) [inline src/k8s-extension/azext_k8s_extension/custom.py:94]: A common convention is to use one option to support both json string and json file. `azure-cli-core` has `type=validate_file_or_dict` for this usage.

File: src/k8s-extension/azext_k8s_extension/_consts_private.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+EXTENSION_NAME = 'k8s-extension-private'
+VERSION = "0.1PP.14"
```
[the remark this answers is not in the document]
Comment (jonathan-innis) [reply src/k8s-extension/azext_k8s_extension/_consts_private.py:8]: Will remove to keep this out of production

File: src/k8s-extension/README.rst [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Microsoft Azure CLI 'k8s-extension' Extension
+=============================================
+
+This package is for the 'k8s-extension' extension.
+i.e. 'az k8s-extension'
```
[the remark this answers is not in the document]
Comment (jonathan-innis) [reply src/k8s-extension/README.rst:5]: Done

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**